src/Altinn.App.Api/Controllers/InstancesController.cs

```diff
@@ -369,6 +370,12 @@ await _instanceClient.DeleteInstance(
             _logger.LogInformation("Events sent to process engine: {Events}", change?.Events);
             await _processEngine.HandleEventsAndUpdateStorage(instance, null, change?.Events);
         }
+        // Let the AltinnAppException through as it is handled in IExceptionHandler
+        catch (AltinnAppException)
```

Comment:
The part I don't like about this is the kind of coupling we get that isn't expressed anywhere (at least not cohesively). IMO it is hard to trace and maintain over time, so I think it's not a good solution for something to be used application-wide. In this particular case, the caller and callee have to be kept in sync, but the compiler doesn't help us :( 

Since this is all internal though, we could let this through as long as we think of it as a temporary solution to this specific problem.

Reply:
Do you object to this specific case of letting an exception through (which I totally agree, but wanted to preserve behavior). Or the general concept of throwing exceptions with explicit problem details info and a central catch middleware?

If this approach is accepted it would obviously cause a bigger change to how we handle errors.

Reply:
I have also purposefully limited the available status codes that can be returned, to reduce surprises for api documentation.